src/Tribe/Models/Post_Types/Event.php

```diff
@@ -64,16 +66,34 @@ protected function build_properties( $filter ) {
 				$duration = $end_date_utc_object->getTimestamp() - $start_date_utc_object->getTimestamp();
 			}
 
-			// An event is multi-day if its end date is after the end-of-day cutoff of the start date.
-			$is_multiday           = $end_of_day_object < $end_date_object;
-			$multiday              = false;
+			/*
+			 * An event is multi-day if its end date is after the end-of-day cutoff of the start date.
+			 * We add one second to make sure events ending at end-of-day, same day, cutoff are not marked as multi-day.
+			 */
+			$is_multiday = $end_of_day_object->add( $one_second ) < $end_date_object;
+			$multiday    = false;
 
 			// Without a context these values will not make sense; we'll set them if the `$filter` argument is a date.
 			$starts_this_week      = null;
 			$ends_this_week        = null;
 			$happens_this_week     = null;
 			$this_week_duration    = null;
 
+			// Multi-day events will span at least two days: the day they start on and the following one.
+			if ( $is_multiday ) {
+				/*
+				 * Count the number of cut-offs happening before the end date and add 1.
+				 * Do not add 1 for all-day events as they span cut-off to cut-off.
+				 */
+				$multiday       = $all_day ? 0 : 1;
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 1 space but found 7 spaces

<sub>hash: 46f6cd9c2ee533aeba02e3743ec3cbf2</sub>